yt_dlp/extractor/genius.py

```diff
@@ -0,0 +1,130 @@
+            'title': 'md5:c6355f7fa8a70bc86492a3963919fc15',
+            'description': 'md5:1774638c31548b31b037c09e9b821393',
+            'tags': 'count:3',
+            'uploader_id': '4863540648001',
+            'duration': 2685.099,
+            'upload_date': '20210909',
+            'timestamp': 1631209167,
+            'thumbnail': r're:^https?://.*\.jpg$',
+        },
+    }]
+
+    def _real_extract(self, url):
+        display_id = self._match_id(url)
+        webpage = self._download_webpage(url, display_id)
+
+        metadata = self._parse_json(self._html_search_regex(
+            r'<meta content="({[^"]*})"', webpage, 'metadata'), display_id)
+        video_id = traverse_obj(
+            metadata, ('video', 'provider_id'),
+            ('dfp_kv', lambda _, x: x['name'] == 'brightcove_video_id', 'values', 0), get_all=False)
+        if not video_id:
+            raise ExtractorError('Brightcove video id not found in webpage')
+
+        config = self._search_json(r'var\s*APP_CONFIG\s*=', webpage, 'config', video_id, default={})
+        account_id = config.get('brightcove_account_id', '4863540648001')
+        player_id = traverse_obj(
+            config, 'brightcove_standard_web_player_id', 'brightcove_standard_no_autoplay_web_player_id',
+            'brightcove_modal_web_player_id', 'brightcove_song_story_web_player_id', default='S1ZcmcOC1x')
+
+        return self.url_result(
+            smuggle_url(
+                f'https://players.brightcove.net/{account_id}/{player_id}_default/index.html?videoId={video_id}',
+                {'referrer': url}), 'BrightcoveNew', video_id)
+
+
+class GeniusLyricsIE(InfoExtractor):
+    _VALID_URL = r'https?://(?:www\.)?genius\.com/(?P<id>[^?/#]+)-lyrics[?/#]?'
+    _TESTS = [{
+        'url': 'https://genius.com/Lil-baby-heyy-lyrics',
+        'playlist_mincount': 2,
+        'info_dict': {
+            'id': '8454545',
+            'title': 'Heyy',
+            'description': 'Heyy by Lil Baby',
+        },
+    }, {
+        'url': 'https://genius.com/Outkast-two-dope-boyz-in-a-cadillac-lyrics',
+        'playlist_mincount': 1,
+        'info_dict': {
+            'id': '36239',
+            'title': 'Two Dope Boyz (In a Cadillac)',
+            'description': 'Two Dope Boyz (In a Cadillac) by OutKast',
+        },
+    }, {
+        'url': 'https://genius.com/Playboi-carti-rip-lyrics',
+        'playlist_mincount': 1,
+        'info_dict': {
+            'id': '3710582',
+            'title': 'R.I.P.',
+            'description': 'R.I.P. by Playboi Carti',
+        },
+    }]
+
+    def _real_extract(self, url):
+        display_id = self._match_id(url)
+        webpage = self._download_webpage(url, display_id)
+
+        escaped_json = self._search_regex(
+            r'window\.__PRELOADED_STATE__\s*=\s*JSON\.parse\((.*)\);', webpage, 'escaped json', display_id)
+        unescaped_json = self._parse_json(escaped_json, display_id, transform_source=js_to_json)
```

Comment:
note to self: see if this can be done with `_search_json`. Ref #5111

Reply:
got it to work with `contains_pattern=r'\'{(?s:.+)}\''`